harvard-university-of-worcester.csl

```diff
@@ -0,0 +1,376 @@
+        <text term="no date"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="container-title">
+    <text variable="container-title" font-style="italic"/>
+  </macro>
+  <macro name="collection-title">
+    <choose>
+      <if type="book chapter" match="any">
+        <text variable="collection-title"/>
+      </if>
+      <else>
+        <text variable="collection-title" font-style="italic"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="series-episode">
+    <choose>
+      <if type="broadcast" match="any">
+        <text variable="collection-number" prefix="Series "/>
+        <choose>
+          <if match="all" variable="number">
+            <text variable="number" prefix=", episode " suffix="."/>
+          </if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="single-quote-title">
+    <text variable="title" prefix=" '" suffix="'."/>
+  </macro>
+  <macro name="translator">
+    <text term="translator" form="short" plural="true" text-case="capitalize-first" suffix="."/>
+    <names variable="translator" prefix=" " suffix="."/>
+  </macro>
+  <macro name="original-title">
+    <text variable="original-title" font-style="italic" prefix="[" suffix="]"/>
+  </macro>
+  <macro name="genre">
+    <text variable="genre"/>
+  </macro>
+  <macro name="volume-number">
+    <number variable="volume"/>
+  </macro>
+  <macro name="issue-number">
+    <number variable="issue"/>
+  </macro>
+  <macro name="event-name">
+    <text variable="event"/>
+  </macro>
+  <macro name="event-place">
+    <text variable="event-place"/>
+  </macro>
+  <macro name="event-date">
+    <date form="text" variable="event-date">
+      <date-part name="day"/>
+      <date-part name="month"/>
+    </date>
+  </macro>
+  <macro name="broadcast-details">
+    <choose>
+      <if type="broadcast interview" match="any">
+        <text variable="publisher" suffix=","/>
+        <date form="text" variable="issued" prefix=" ">
+          <date-part name="day"/>
+          <date-part name="month"/>
+        </date>
+      </if>
+    </choose>
+  </macro>
+  <macro name="scale">
+    <text variable="scale"/>
+  </macro>
+  <macro name="director">
+    <text term="director" prefix=":"/>
+    <names variable="director"/>
+  </macro>
+  <macro name="graph-or-figure">
+    <choose>
+      <if type="graphic" match="any">
+        <text value="graph"/>
+      </if>
+      <else-if type="figure" match="any">
+        <text value="fig"/>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="day-month">
+    <choose>
+      <if type="webpage interview personal_communication post-weblog" match="any">
+        <date form="text" date-parts="year-month-day" variable="issued">
+          <date-part name="day"/>
+          <date-part name="month"/>
+        </date>
+      </if>
+    </choose>
+  </macro>
+  <macro name="interviewer">
+    <text value="Interviewed by "/>
+    <names variable="interviewer"/>
+    <text value=" for "/>
```

Comment:
haven't followed the logic here entirely, but wouldn't it make more sense to pack the "for" into a group delimiter, in case you have an interview without a magazine/TV station?